### PR TITLE
[DOC] Fix rendering and add some refs in `region_extraction.rst`

### PR DESCRIPTION
**Pull request by NicolasGensollen**
Small PR fixing some rendering issues and adding references to API and Glossary.

### CODE REVIEW

File: doc/connectivity/region_extraction.rst [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+================================================
 
 .. topic:: **Page summary**
 
-   This section shows how to use Region Extractor to extract brain connected
-   regions/components into a separate brain activation region and also
-   shows how to learn functional connectivity interactions between each
-   separate region.
+   This section shows how to use :class:`~nilearn.regions.RegionExtractor`
+   to extract brain connected regions/components into a separate brain
```
Comment (bthirion) [inline doc/connectivity/region_extraction.rst:10]:
```suggestion
   to extract connected regions/components into a separate brain
```

File: doc/connectivity/region_extraction.rst [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+   activation region and also shows how to learn functional connectivity
```
Comment (bthirion) [inline doc/connectivity/region_extraction.rst:11]:
```suggestion
    region and also shows how to learn functional connectivity
```

**Review by bthirion — COMMENTED** [review]
Thx for cleaning that one.

**Comment by NicolasGensollen** [comment]
[Link](https://8786-1235740-gh.circle-artifacts.com/0/dev/connectivity/region_extraction.html) to the page in the artifacts.

File: examples/03_connectivity/plot_extract_regions_dictlearning_maps.py [hunk 3]
```diff
@@ -114,7 +119,7 @@
 
 ###############################################################################
 # Plot resulting connectomes
-# ----------------------------
+# --------------------------
```
Comment (bthirion) [inline examples/03_connectivity/plot_extract_regions_dictlearning_maps.py:122]: This line does not render well...

**Review by bthirion — COMMENTED** [review]
LGTM otherwise

File: examples/03_connectivity/plot_extract_regions_dictlearning_maps.py (hunk 3 above)
Comment (NicolasGensollen) [reply examples/03_connectivity/plot_extract_regions_dictlearning_maps.py:122]: What do you mean? 
It seems to render fine in the artifacts:

ex

Comment (bthirion) [reply examples/03_connectivity/plot_extract_regions_dictlearning_maps.py:122]: OK sorry, the lines do not render greatly on https://8786-1235740-gh.circle-artifacts.com/0/dev/connectivity/region_extraction.html but maybe this simply comes from the rst rendering on this particular page.

**Comment by NicolasGensollen** [comment]
Direct links to the newest artifacts:

- [user guide](https://8791-1235740-gh.circle-artifacts.com/0/dev/connectivity/region_extraction.html)
- [example](https://8792-1235740-gh.circle-artifacts.com/0/dev/auto_examples/03_connectivity/plot_extract_regions_dictlearning_maps.html#sphx-glr-auto-examples-03-connectivity-plot-extract-regions-dictlearning-maps-py)

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Comment by NicolasGensollen** [comment]
Thanks! Merging...

**Pull request merged**